packages/@headlessui-react/src/hooks/use-transition.ts

```diff
@@ -4,6 +4,34 @@ import { useDisposables } from './use-disposables'
 import { useFlags } from './use-flags'
 import { useIsoMorphicEffect } from './use-iso-morphic-effect'
 
+if (
+  typeof process !== 'undefined' &&
+  typeof globalThis !== 'undefined' &&
```

Comment:
should we also check for `typeof process.env !== 'undefined'` _or_ make the `process.env[]` lookup an optional chain, just to prevent a crash in some very weird setups:

Reply:
Yeah not a bad idea honestly